File: Dataset/Leetcode/valid/6/159.cpp
 class Solution {
public:
    string XXX(string s, int numRows) {
        string result = "";
        if(numRows == 1 || s.size() <= numRows) return s;
        for(int i = 0;i < numRows;i++)
        {
            int start = i;
            int step;
            if((i == numRows-1) or (i == 0))
            {
                step = 2 * (numRows - 1);
                while(start < s.size())
                {
                result += s[start];
                start += step;
                }
            }
            else
            {
                step = 2 * (numRows -1 -i);
                while(start < s.size())
                {
                result += s[start];
                start += step;
                step = 2* numRows -step-2;
                }
            }
            
        }

        return result;
    }
};

File: Dataset/Leetcode/train/66/159.java
 class Solution {
    public int[] XXX(int[] digits) {
        int n = digits.length;
        for(int i = n - 1; i >= 0; i--){
            digits[i] += 1;
            if(digits[i] > 9)
                digits[i] = 0;
            else
                break;
        }
        if(digits[0] == 0){ // 避免[9,9,9]情况，边界问题，扩充数组
            digits = new int[n + 1];
            digits[0] = 1;
        }
        return digits; 
    }
}

